relayer/pkg/chainlink/keys/keystore_test.go

```diff
@@ -0,0 +1,110 @@
+package keys_test
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math/big"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/smartcontractkit/caigo"
+	"github.com/smartcontractkit/chainlink-starknet/relayer/pkg/chainlink/keys"
+	"github.com/stretchr/testify/require"
+)
+
+func TestKeyStoreAdapter(t *testing.T) {
+	var (
+		getter = newMemKeyGetter()
+
+		starknetPK         = generateTestKey(t)
+		starknetSenderAddr = "legit"
+	)
+	getter.Put(starknetSenderAddr, starknetPK)
+
+	lk := keys.NewLooppKeystore(getter)
+	adapter := keys.NewKeystoreAdapter(lk)
+	// test that adapter implements the loopp spec. signing nil data should not error
+	// on existing sender id
+	signed, err := adapter.Loopp().Sign(context.Background(), starknetSenderAddr, nil)
+	require.Nil(t, signed)
+	require.NoError(t, err)
+
+	signed, err = adapter.Loopp().Sign(context.Background(), "not an address", nil)
+	require.Nil(t, signed)
+	require.Error(t, err)
+
+	hash, err := caigo.Curve.PedersenHash([]*big.Int{big.NewInt(42)})
+	require.NoError(t, err)
+	r, s, err := adapter.Sign(context.Background(), starknetSenderAddr, hash)
+	require.NoError(t, err)
+	require.NotNil(t, r)
+	require.NotNil(t, s)
+
+	pubx, puby, err := caigo.Curve.PrivateToPoint(starknetPK)
+	require.NoError(t, err)
+	require.True(t, caigo.Curve.Verify(hash, r, s, pubx, puby))
+}
+
+func generateTestKey(t *testing.T) *big.Int {
+	// sadly generating a key can fail, but it should happen infrequently
+	// best effort here to  avoid flaky tests
+	var generatorDuration = 1 * time.Second
+	d, exists := t.Deadline()
+	if exists {
+		generatorDuration = time.Until(d) / 2
+	}
+	timer := time.NewTicker(generatorDuration)
+	defer timer.Stop()
+	var key *big.Int
+	var err error
+
+GENERATEKEY:
+	for {
+		select {
+		case <-timer.C:
+			key = nil
+			err = fmt.Errorf("failed to generate test key in allotted time")
+			break GENERATEKEY
+		default:
+			key, err = caigo.Curve.GetRandomPrivateKey()
+			if err == nil {
+				break GENERATEKEY
+			}
+		}
+	}
```

Comment:
WDYT about trying to avoid the label somehow?
```suggestion
	generate := func() bool {
		select {
		case <-timer.C:
			key = nil
			err = fmt.Errorf("failed to generate test key in allotted time")
			return true
		default:
			key, err = caigo.Curve.GetRandomPrivateKey()
			if err == nil {
				return true
			}
		}
		return false
	}
	for !generate() { }
```

Reply:
sure.

what is the objection about the named loop?

Reply:
Just that they are generally more difficult to reason about and prone to errors in future edits. Or sometimes a code smell that there is too much nesting/indentation regardless.
I usually find alternative styles to be simpler anyways, except maybe in cases of tight double for loops?